api/src/main/java/org/openmrs/module/pihcore/event/PatientUpdateEventConsumer.java

```diff
@@ -57,6 +57,11 @@ public void startup() {
         try {
             statusDb = new Rocks(new File(config.getContext().getModuleDataDir(), "status.db"));
             snapshotInitialized = BooleanUtils.isTrue(statusDb.get("snapshotInitialized"));
+            if (!snapshotInitialized) {
+                performInitialSnapshot();
+                statusDb.put("snapshotInitialized", Boolean.TRUE);
+                snapshotInitialized = true;
+            }
```

Comment:
I assume you determined it's okay to run this during startup now?

Reply:
Hard to know.  It will block startup until it completes, so that's the main factor - how long it takes on a system that hasn't run it before.  Maybe we should kick it off in a separate thread...

Reply:
Oh, right, I had forgotten the reasoning, I was thinking it was failing during startup... yeah, we can take a look at performance and decide, should keep it in mind the first time we roll out to a system with a large DB... a one-time 5 minute run seems fine to me, but we should remember not to panic the time that deployment starts... actually, thinking out probably worth putting a log.info or log.warn this to remind us if we are "log-watching" a prod startup.

Reply:
Ok, I just pushed a new commit that runs this setup in a separate thread and adds some additional logging as well

Reply:
Was trying to look at the new commit but Github appears a little flaky right now...the only issue with a separate thread, do we potentially have a risk of an issue if updates start coming in while it's initializing?  Also, do we need to makes sure this is initialized before any background tasks (like close stale visits?) runs?

Reply:
I don't think those things are issues.  But there is perhaps an issue with timing where there could be a small gap between the time when the initialization code runs and the binlog monitoring starts, during which time changes could happen that are not detected.  Let's discuss.